articles/proofs-arguments-and-zero-knowledge/facts/Schwartz-Zippel-Lemma.md

```diff
@@ -0,0 +1,4 @@
+### Schwartz-Zippel Lemma
+
+**Lemma:
+Let $F$ be any field, and let g: $F^m → F$ be a nonzero $m$-variate polynomial of total degree at most $d$. Then on any finite set $S \subseteq F$ : $$Pr_{x←S^m} [g(x) = 0] ≤ d/|S|.$$
```

Comment:
I don't understand the math equation, can you provide a text version of the equation below?

Reply:
F^m -> F means:
we have m different variables and each of them in F -> F^m.
(-> F) means the result in F.


x <- S^m means choose x of (x_0,.., x_m)

Reply:
The total degree of m-variate polynomial (d) is the number of possible root. Hence we have d in the equation

Reply:
|S| means the number of different possible values of S